src/java/org/apache/cassandra/utils/memory/BufferPool.java

```diff
@@ -232,8 +236,8 @@ private Chunk allocateMoreChunks()
                     if (MEMORY_USAGE_THRESHOLD > 0)
                     {
                         noSpamLogger.info("Maximum memory usage reached ({}), cannot allocate chunk of {}",
-                                          prettyPrintMemory(MEMORY_USAGE_THRESHOLD),
-                                          prettyPrintMemory(MACRO_CHUNK_SIZE));
+                                          () -> params(prettyPrintMemory(MEMORY_USAGE_THRESHOLD),
+                                                       prettyPrintMemory(MACRO_CHUNK_SIZE)));
```

Comment:
@jonmeredith Possibly naive question...why don't we just pre-compute human readable versions of `MEMORY_USAGE_THRESHOLD` and `MACRO_CHUNK_SIZE` (using `prettyPrintMemory()`)? There are some tests that write to `MEMORY_USAGE_THRESHOLD` of course, but we can get a `static` setter in place that updates its "companion" as well.

Reply:
I think pre-computing them is a good plan where constant or they are set infrequently.  The `sizeInBytes()` in the failed allocation will be variable, but fixing all the other occurrences would definitely be a good thing.